lib/tasks/db.rake

```diff
@@ -48,6 +48,17 @@ namespace :db do
     end
   end
 
+  task :post_migration_hook do
+    at_exit do
+      unless %w(C POSIX).include?(ActiveRecord::Base.connection.execute('SELECT datcollate FROM pg_database WHERE datname = current_database();').first['datcollate'])
```

Comment:
OK, so this would throw this warning on my server every time I run db:migrate, even though I have not updated glibc after indexing. This will probably scare a lot of people. We need to come up with something that only warns you if you actually have an issue.

Reply:
Unfortunately, I think the only reliable way to check that there is actually an issue is to use `amcheck`. However, there are two issues with that:
1. it will impact performances for a few seconds to a few minutes
2. it requires superuser access to the database, which Mastodon typically doesn't have

Reply:
We *could* remove that altogether and instead just rely on the documentation and release notes… maybe catch some specific migration errors that are likely to be caused by this… but I'm afraid those solutions would not be enough to convince people to look into silent corruptions…